generate-errors/src/lib.rs

```diff
@@ -1,114 +1,174 @@
+    if !bevy_errors_path.try_exists()? {
+        return Err(anyhow!("The path ({bevy_errors_path:?}) is invalid"));
+    }
 
-#[derive(Serialize)]
-pub struct FrontMatterErrorCode {
-    pub title: String,
-    pub weight: usize,
-}
+    let mut error_page_paths: Vec<PathBuf> = vec![];
 
-impl From<&ErrorCode> for FrontMatterErrorCode {
-    fn from(asset: &ErrorCode) -> Self {
-        FrontMatterErrorCode {
-            title: asset.code.clone(),
-            weight: 0,
-        }
-    }
-}
+    let entries = bevy_errors_path.read_dir()?;
+    let regex = Regex::new(r"B[0-9]{4}")?;
 
-fn visit_dirs(dir: PathBuf, section: &mut Section) -> io::Result<()> {
-    assert!(dir.is_dir(), "The path to the errors is not a directory");
-    let error_code_pattern = Regex::new(r"B[0-9]{4}\.md").unwrap();
-    for entry in fs::read_dir(dir)? {
+    for entry in entries {
+        // You can't propagate a value multiple times
+        // due to values moving, so it has to be put
+        // in a variable to be used multiple times.
         let entry = entry?;
-        let path = entry.path();
-        if !path.is_dir() {
-            let file_name = path
-                .file_name()
-                .unwrap()
-                .to_os_string()
-                .into_string()
-                .unwrap();
-            if !error_code_pattern.is_match(&file_name) {
-                continue;
-            }
-
-            let error_code = read_to_string(path.clone())?;
-
-            let code = file_name.trim_end_matches(".md").to_owned();
-            section.content.push(ErrorCode {
-                content: error_code
-                    .trim_start_matches(&format!("# {}", code.clone()))
-                    .replace("```rust,*", "```rust")
-                    .lines()
-                    .map(|line| {
-                        // throw away `should_panic` and `no_run` to fix code highlighting
-                        if line.starts_with("```rust,") {
-                            "```rust"
-                        } else {
-                            line
-                        }
-                    })
-                    .collect::<Vec<&str>>()
-                    .join("\n"),
-                code,
-            });
+
+        if entry.metadata()?.is_dir() {
+            continue;
+        }
+
+        if entry
+            .file_name()
+            .to_str()
+            .is_some_and(|value| regex.is_match(value))
+        {
+            error_page_paths.push(entry.path());
         }
     }
 
-    section.content.sort_by(|a, b| a.code.cmp(&b.code));
+    let mut results_map: HashMap<String, String> = HashMap::new();
 
-    Ok(())
+    for path in error_page_paths {
+        let file_name: String = path
+            .file_name()
+            .ok_or(anyhow!("An error page path has an invalid file stem"))?
+            .to_string_lossy()
+            .into_owned();
+        let content = String::from_utf8(fs::read(path)?)?;
```

Comment:
Looking with fresh eyes, i probably should've just used `fs::read_from_string`, will fix later.